pagopa/io-app
# feat: [IOCOM-364,IOCOM-234] Optimisations on multiple renderings of messages' section

## Short description
This PR optimises the rendering process of the messages' section.

## List of changes proposed in this pull request
- `MessagesHomeScreen`: 
  - `createSelector` moved to a dedicated function in order to avoid returning a new function instance every time that `mapStateToProps` was called. 
  - `messageSectionStatusActive` changed to a boolean in order to avoid receiving a new `Option` instance every time that the backed status was updated
- `MessageList/index`:
  - removed the re-mapping of `paginatedState` in order to avoid the creation of a new instance every time that `mapStateToProps` was called
  - `isNoticePaid` moved into the Item component in order not to create a new function every time that `mapStateToProps` was called

## How to test
Using the `io-dev-api-server`, run the application and go to both Inbox and Archived messages. Switch bottom tabs and check that the message section is not rendering its content (unless a refresh timeout is hit, or something that may invoke a message reload is triggered - like receiving a push notification)
